### PR TITLE
Update to typescript 5  + mobx

### PR DESCRIPTION
**Pull request by lbwexler**
Adding additional minor dependency upgrades here as well

Interestingly, we were running to a bit of an issue trying to upgrade mobx-react-lite to the next major version.  Looking into that.   

Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [X] Caught up with `develop` branch as of last change.
- [X] Added CHANGELOG entry, or determined not required.
- [X] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [X] Updated doc comments / prop-types, or determined not required.
- [X] Reviewed and tested on Mobile, or determined not required.
- [X] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,13 +4,18 @@
 
 ### 💥 Breaking Changes
 
-* `DataViewConfig` no longer directly supports the passing of `GridConfig` parameters.
-  Instead, all direct `GridConfig` toggles must be passed through the new `gridOptions`
-  parameter.
+* Apps should update their Typescript dependency to v5.1. This should be a drop-in for most
+  applications, or require only minor changes. Note that Hoist has not yet adopted the updated
+  approach to decorators added in TS v5, maintaining compatibility with the "legacy" syntax.
```
Comment (amcclain) [inline CHANGELOG.md:9]: @lbwexler does this look like a useful / correct description of the TS upgrade from an app's POV?

File: package.json [hunk 2]
```diff
@@ -99,8 +99,8 @@
     "react-dom": "^18.2.0",
     "stylelint": "15.x",
     "stylelint-config-standard-scss": "10.x",
-    "type-fest": "^3.10.0",
-    "typescript": "~4.9.5"
+    "type-fest": "4.x",
+    "typescript": "~5.1.6"
```
Comment (amcclain) [inline package.json:103]: Do we want to fix on TS 5.1, vs allowing 5.x?  I see that 5.2 is currently in beta, but don't know how big of a deal TS "minor" version updates typically are.

File: package.json [hunk 3]
```diff
@@ -56,8 +56,8 @@
     "inter-ui": "~3.19.3",
     "lodash": "~4.17.21",
     "lodash-inflection": "~1.5.0",
-    "mobx": "~6.8.0",
-    "mobx-react-lite": "~3.4.0",
+    "mobx": "~6.9.1",
+    "mobx-react-lite": "~3.4.3",
```
Comment (amcclain) [inline package.json:60]: Note that this lib has moved to 4.x releases - worth looking at those updates in this branch?

File: yarn.lock [hunk 4]
```diff
@@ -7055,11 +7055,16 @@ typed-styles@^0.0.7:
   resolved "https://registry.yarnpkg.com/typed-styles/-/typed-styles-0.0.7.tgz#93392a008794c4595119ff62dde6809dbc40a3d9"
   integrity sha512-pzP0PWoZUhsECYjABgCGQlRGL1n7tOHsgwYv3oIiEpJwGhFTuty/YNeduxQYzXXa3Ge5BdT6sHYIQYpl4uJ+5Q==
 
-typescript@^4.9, typescript@~4.9.5:
+typescript@^4.9:
```
Comment (amcclain) [inline yarn.lock:7058]: I am prepping an update to our shared eslint config and dev-utils packages to update to TS 5 also, so we don't still have TS 4 in the mix!

**Review by amcclain — COMMENTED** [review]
Eager to get this in - working on the dev-utils side.  Please see minor questions below.

**Comment by lbwexler** [comment]
Meant to mention -- I looked at the upgrade to mobx-react-lite to v 4.x and
there were issues with it   Will look again, but I think we need to hold
off -- its brand new.

File: yarn.lock (hunk 4 above)
Comment (amcclain) [reply yarn.lock]: Updated @xh/hoist-dev-utils to newly built v7-SNAP, includes transitive dep on updated @xh/eslint-config v6 with TS 5.x support.

File: cmp/grid/columns/Column.ts [hunk 5]
```diff
@@ -455,7 +462,7 @@ export class Column {
     autosizeBufferPx: number;
     autoHeight: boolean;
     editable: boolean | ColumnEditableFn;
-    editor: ColumnEditorFn;
```
Comment (amcclain) [inline cmp/grid/columns/Column.ts:458]: Thanks for fixing this!

**Review by amcclain — APPROVED** [review]

File: cmp/grid/columns/Column.ts (hunk 5 above)
Comment (lbwexler) [reply cmp/grid/columns/Column.ts:458]: don't thank me, thank typescript -- it insisted!

File: CHANGELOG.md (hunk 1 above)
Comment (lbwexler) [reply CHANGELOG.md:9]: 👍🏻

**Pull request merged**